proto/cosmos/crypto/secp256k1/keys.proto

```diff
@@ -11,12 +12,27 @@ option go_package = "github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1";
 // the x-coordinate. Otherwise the first byte is a 0x03.
 // This prefix is followed with the x-coordinate.
 message PubKey {
+  option (legacy_amino.name) = "tendermint/PubKeySecp256k1";
+  // The Amino encoding is simply the inner bytes field, and not the Amino
+  // encoding of the whole PubKey struct.
+  //
+  // Example (JSON):
+  // s := PubKey{Key: []byte{0x01}}
+  // out := AminoJSONEncoder(s)
+  //
+  // Then we have:
+  // out == `"MQ=="`
+  // out != `{"key":"MQ=="}`
+  option (legacy_amino.encoding)      = "bytes field";
```

Comment:
same question again regarding `(legacy_amino.encoding)` at message level.

Reply:
I would imagine something more like:

```proto
message PubKey {
   option (legacy_amino.name) = "tendermint/PubKeySecp256k1";
   option (gogoproto.goproto_stringer) = false;
   bytes key = 1 [
      (legacy_amino.encoding) = 'specialCodegenHint'
   ];
}
```

Reply:
There are cases where it isn't a particular field that has a custom encoding, but the whole message. See [example](https://github.com/amaurym/go-amino-spec/blob/ccf6e13eb07f584e19456e043a75f40aed0dc1f6/spec_test.go#L36-L41)